l5kit/l5kit/data/map_api.py

```diff
@@ -134,6 +144,92 @@ def get_lane_coords(self, element_id: str) -> dict:
 
         return {"xyz_left": xyz_left, "xyz_right": xyz_right}
 
+    @staticmethod
+    def interpolate(xyz: np.ndarray, cum_dist: np.ndarray, step: float, method: InterpolationMethod) -> np.ndarray:
+        """
+        Interpolate points based on cumulative distances from the first one. Two modes are available:
+        INTER_METER: interpolate using step as a meter value over cumulative distances (variable len result)
+        INTER_ENSURE_LEN: interpolate using a variable step such that we always get step values
+        Args:
+            xyz (np.ndarray): XYZ coords
+            cum_dist (np.ndarray): distances from the first coords of xyz. Same length as xyz
+            step (float): param for the interpolation
+            method (InterpolationMethod): method to use to interpolate
+
+        Returns:
+            np.ndarray: the new interpolated coordinates
+        """
+        if method == InterpolationMethod.INTER_ENSURE_LEN:
+            step = int(step)
+            assert step > 1, "step must be at least 2 with INTER_ENSURE_LEN"
+            steps = np.linspace(cum_dist[0], cum_dist[-1], step)
+
+        elif method == InterpolationMethod.INTER_METER:
+            assert step > 0, "step must be greater than 0 with INTER_FIXED"
+            steps = np.arange(cum_dist[0], cum_dist[-1], step)
+        else:
+            raise NotImplementedError("interpolation method unknown")
+
+        xyz_inter = np.zeros((len(steps), 3), dtype=xyz.dtype)
+        xyz_inter[:, 0] = np.interp(steps, xp=cum_dist, fp=xyz[:, 0])
+        xyz_inter[:, 1] = np.interp(steps, xp=cum_dist, fp=xyz[:, 1])
+        xyz_inter[:, 2] = np.interp(steps, xp=cum_dist, fp=xyz[:, 2])
```

Comment:
I'm not sure about it but I think that the `np.interp` would allow you to do this on a single call, but have to double check.

Reply:
according to the docstring it's only 1D and it doesn't take axis as an argument :(